src/async-wrap.cc

```diff
@@ -13,16 +14,91 @@ using v8::Function;
 using v8::FunctionCallbackInfo;
 using v8::Handle;
 using v8::HandleScope;
+using v8::HeapProfiler;
 using v8::Integer;
 using v8::Isolate;
 using v8::Local;
 using v8::Object;
+using v8::RetainedObjectInfo;
 using v8::TryCatch;
 using v8::Value;
 using v8::kExternalUint32Array;
 
 namespace node {
 
+static const char* const provider_names[] = {
+#define V(PROVIDER)                                                           \
+  #PROVIDER,
+  NODE_ASYNC_PROVIDER_TYPES(V)
+#undef V
+};
+
+
+class RetainedAsyncInfo: public RetainedObjectInfo {
+ public:
+  explicit RetainedAsyncInfo(uint16_t class_id, Handle<Value> wrapper);
```

Comment:
`HeapProfiler::WrapperInfoCallback` seems to expect a `Handle<Value>`. Compiler gives me
```
type mismatch at 2nd parameter ('Handle<v8::Value>' vs 'Local<v8::Value>')
```